src/vm/functions/database.rs

```
@@ -125,3 +127,56 @@ pub fn special_delete_entry(args: &[SymbolicExpression],
 
     env.global_context.database.delete_entry(&env.contract_context.name, map_name, &key)
 }
+
+pub fn special_get_block_info(args: &[SymbolicExpression], 
+                              env: &mut Environment, 
+                              context: &LocalContext) -> Result<Value> {
+
+    // (get-block-info property-name block-height-int)
+
+    if args.len() != 2 {
+        return Err(Error::new(ErrType::InvalidArguments(
+            "(get-block-info ...) requires at least 2 arguments: the block property name and the block height expression".to_string())))
+    }
+
+    // Handle the block property name input arg.
+    let property_name = args[0].match_atom()
+        .ok_or(Error::new(ErrType::InvalidArguments("First argument to (get-block-info ...) must be block property name".to_string())))?;
+
+    let block_info_prop = BlockInfoProperty::from_str(property_name)
+        .ok_or(Error::new(ErrType::InvalidArguments("First argument to (get-block-info ...) is not a valid block property name".to_string())))?;
+
+
+    // Handle the block-height input arg clause.
+    let height_eval = eval(&args[1], env, context)?;
+    let height_value = match height_eval {
```

This should throw an error if  `height_value` is negative.